src/flow/src/utils.rs

```diff
@@ -176,15 +183,85 @@ impl Arrangement {
         Ok(max_late_by)
     }
 
+    /// find out the time of next update in the future
+    pub fn get_next_update_time(&self) -> Option<Timestamp> {
+        let next_batch = {
+            let mut iter = self.spine.iter();
+            if self.full_arrangement {
+                iter.next();
+                iter
+            } else {
+                iter
+            }
```

Comment:
can you add a comment to explain why?

Reply:
> can you add a comment to explain why?

refactor a bit, now require passing a `now` timestamp